Ginger/Ginger/Actions/_Common/ActUIElementLib/ActUIElementEditPage.xaml.cs

```diff
@@ -195,89 +229,246 @@ void ShowPlatformSpecificPage()
+                    break;
+
+                case eElementAction.SelectByIndex:
+                case eElementAction.SetSelectedValueByIndex:
+                    if (mAction.ElementType != eElementType.RadioButton)
+                    {
+                        if (mAction.ElementLocateBy == eLocateBy.POMElement)
+                        {
+                            elementList.Add(GetPomOptionalValuesComboBox(ActUIElement.Fields.ValueToSelect, ePomElementValuesType.Indexs));
+                        }
+                        else
+                        {
+                            possibleValues = String.IsNullOrEmpty(mAction.GetInputParamValue(ActUIElement.Fields.ValueToSelect)) ? new List<string>() { "0", "1", "2", "3", "4", "5", "6", "7", "8", "9", "10" } :
+                                                   new List<string>() { mAction.GetInputParamValue(ActUIElement.Fields.ValueToSelect) };
+
+                            elementList.Add(GetElementConfigControl("Value", Fields.ValueToSelect, eElementType.ComboBox, possibleValues));
+
+                        }
+                    }
+                    break;
+
+                case eElementAction.Click:
+                case eElementAction.AsyncClick:
+                case eElementAction.DoubleClick:
+                    if (mAction.ElementType == eElementType.MenuItem || mAction.ElementType.Equals(eElementType.TreeView))
+                    {
+                        possibleValues = String.IsNullOrEmpty(mAction.GetInputParamValue(ActUIElement.Fields.ValueToSelect)) ? new List<string>() { "" } :
+                                               new List<string>() { mAction.GetInputParamValue(ActUIElement.Fields.ValueToSelect) };
+
+                        elementList.Add(GetElementConfigControl("Value", Fields.ValueToSelect, eElementType.TextBox, possibleValues));
+                    }
+                    break;
+
+                case eElementAction.GetControlProperty:
+                    //TODO: find a better way to bind list of enum with possible values.
+                    List<ActUIElement.eElementProperty> propertyList = mPlatform.GetPlatformElementProperties();
+                    List<string> propertyListString = new List<string>();
+                    foreach (ActUIElement.eElementProperty property in propertyList)
+                    {
+                        propertyListString.Add(property.ToString());
+                    }
+
+                    elementList.Add(GetElementConfigControl("Name", Fields.ValueToSelect, eElementType.ComboBox, propertyListString));
+                    break;
+
+                default:
+                    Reporter.ToLog(eLogLevel.DEBUG, mAction.ElementAction.ToString() + "not required config page.");
+                    break;
+            }
+
+            //if widgets and element action is TriggerJavaScriptEvent
+            if (mAction.ElementAction.Equals(eElementAction.TriggerJavaScriptEvent))
```

Comment:
Trigger java script event is only needed for Java driver. So this code is not generic and should not be here.
Why we can not add this check box dynamically ? like other controls